Documentation: add DSA device plugin documentation (#310)
intel/intel-technology-enabling-for-openshift: device_plugins/deploy_dsa.md

5.	Optional: If you want to make any customizations, select YAML view and edit the details. When you are done, click **Create**.

## Verify via web console
1.	Verify CR by checking the status of **Workloads** -> **DaemonSet** -> **intel-dsa-plugin**.
2.	Now `DsaDevicePlugin` is created.

## Create CR via CLI
Apply the CR yaml file:
```
$ oc apply -f https://raw.githubusercontent.com/intel/intel-technology-enabling-for-openshift/main/device_plugins/dsa_device_plugin.yaml
```

## Verify via CLI
Verify that the device plugin CR is ready: 
```
$ oc get DsaDevicePlugin
```
Output: 
```
NAME		        DESIRED		READY	NODE SELECTOR	                                    AGE
dsadeviceplugin-sample  3 	        3       {"intel.feature.node.kubernetes.io/dsa":"true"}     98m
```

# Verify DSA Device Plugin 
After the plugin is deployed, use below command to verify DSA resources: 
```
$ oc describe node srf-2 | grep dsa.intel.com
  dsa.intel.com/wq-user-dedicated:  0
  dsa.intel.com/wq-user-shared:     160
  dsa.intel.com/wq-user-dedicated:  0
  dsa.intel.com/wq-user-shared:     160
 ```

## DSA Resource Configuration

> By default the DSA plugin uses [this configuration file](https://github.com/intel/intel-device-plugins-for-kubernetes/blob/main/demo/dsa.conf).

Reviewer: Could you briefly explain the DSA resource Configuration here, and also add the link point to the detailed explaining?

Reviewer 2: I agree it makes sense to add a bit more on _how_ the configuration is done. @mregmi has some basics already covered . Essentially, the initcontainer just comes with `accel-config` tool that is used to configure the DSA WQs. The "default" setting creates dedicated WQs for each DSA device so that it's four groups per device where each groups is with 1 WQ linked to 1 (dedicated) engine (i.e., each k8s resource is "equal").

I'd also add tha node specific configs are allowed. The default config can be overridden using the configmap where the entries are `dsa.conf` or `dsa-<nodename>.conf`(or both).

At the moment `idxd-init.sh` only assumes that each physical DSA device is configured the same way...

Author: i will add more details here. Thanks

Author: added thanks

Reviewer: @mythi  could you please help review this updated information about the resource configuration?

Author: applied all suggestions

```
$ oc get DsaDevicePlugin
```
Output: 
```
NAME		        DESIRED		READY	NODE SELECTOR	                                    AGE
dsadeviceplugin-sample  3 	        3       {"intel.feature.node.kubernetes.io/dsa":"true"}     98m
```

# Verify DSA Device Plugin 
After the plugin is deployed, use below command to verify DSA resources: 
```
$ oc describe node srf-2 | grep dsa.intel.com
  dsa.intel.com/wq-user-dedicated:  0
  dsa.intel.com/wq-user-shared:     160
  dsa.intel.com/wq-user-dedicated:  0
  dsa.intel.com/wq-user-shared:     160
 ```

## DSA Resource Configuration
By default the DSA plugin uses [this configuration file](https://github.com/intel/intel-device-plugins-for-kubernetes/blob/main/demo/dsa.conf).
The dsa init container comes with a utility called `accel-config` which takes file as input and configures the DSA hardwares based on that.
The default configuration has creates dedicated WQs for each DSA device so that it's four groups per device where each groups is with 1 WQ linked to 1 engine.
Users can customise the config and can use the pre-customised config for their specific use case from [here](https://github.com/intel/idxd-config/tree/stable/contrib/configs)

> There's also a possibility for a node specific congfiguration through passing a nodename via NODE_NAME into initcontainer's environment and passing a node specific profile via configMap volume mount.

Reviewer 2: `NODE_NAME` is passed automatically by the operator so that is not needed in the sentence:
```suggestion
There's also a possibility for a node specific configuration by passing a node specific profile via configMap volume mount.
```

Users can use the steps below to customize the DSA resource configuration:  
1. Create the configmap for DSA resource configuration 
    ```

>     $ oc create configmap --namespace=openshift-operators intel-dsa-config --from-file=<your config file> 

Reviewer 2: minor nit that the default setup script is sensitive about what the file name is:
```suggestion
    $ oc create configmap --namespace=openshift-operators intel-dsa-config --from-file=dsa[-$NODE_NAME].conf 
```
